src/runner/settings_window.cpp

```diff
@@ -106,7 +107,14 @@ std::optional<std::wstring> dispatch_json_action_to_module(const json::JsonObjec
                 }
                 else if (action == L"check_for_updates")
                 {
-                    CheckForUpdatesCallback();
+                    if (!isUpdateCheckThreadRunning)
```

Comment:
Maybe we could use [compare_exchange_strong](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange) in these cases, even though the probability of the bug is really small.

Reply:
compare_exchange_strong added.